lib/lib.es2017.object.d.ts

```diff
@@ -42,4 +42,10 @@ interface ObjectConstructor {
       * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
       */
     entries(o: any): [string, any][];
+
+    /**
+      * Returns an object containing all own property descriptors of an object
+      * @param o Object that contains the properties and methods. This can be an object that you created or an existing Document Object Model (DOM) object.
+      */
+    getOwnPropertyDescriptors<T>(o: T): { [P in keyof T]: PropertyDescriptor };
```

Comment:
`Object.getOwnPropertyDescriptors` output (just like `Object.keys`) should not only be limitied to `keyof`. please see https://github.com/Microsoft/TypeScript/pull/12253 and https://github.com/Microsoft/TypeScript/pull/13971 for relevant discussion.

I would recommend using an index singature along with the mapped type. 

Also consider using `TypedPropertyDescriptor` instead to get better type flow.

```ts
getOwnPropertyDescriptors<T>(o: T): {[P in keyof T]: TypedPropertyDescriptor<T[P]>} & { [x: string]: PropertyDescriptor };
```